### PR TITLE
PSP-9493 | Added purpose on edit and view for document pages

### PR DESCRIPTION
**Pull request by FuriousLlama**

### CODE REVIEW

File: source/frontend/src/features/documents/documentDetail/DocumentDetailForm.tsx [hunk 1]
```diff
@@ -58,6 +60,36 @@ export const DocumentDetailForm: React.FunctionComponent<
     [props.documentTypes],
   );
 
+  const [documentTypePurpose, setDocumentTypePurpose] = useState(null);
+
+  const matchDocumentType = useCallback(
+    (documentTypeId: number) => {
+      const purpose = props.documentTypes.find(
```
Comment (devinleighsmith) [inline source/frontend/src/features/documents/documentDetail/DocumentDetailForm.tsx:67]: Hmm, you need to do a type conversion of documentTypeId even though it is already typed as a number?

Comment (FuriousLlama) [reply source/frontend/src/features/documents/documentDetail/DocumentDetailForm.tsx:67]: good catch, I think I reused the function from a place where the doctype was a string (the form)

File: source/frontend/src/features/documents/documentDetail/DocumentDetailForm.tsx [hunk 2]
```diff
@@ -58,6 +60,36 @@ export const DocumentDetailForm: React.FunctionComponent<
...
+        x => x.id === Number(documentTypeId),
+      )?.documentTypePurpose;
+
+      setDocumentTypePurpose(purpose);
```
Comment (devinleighsmith) [inline source/frontend/src/features/documents/documentDetail/DocumentDetailForm.tsx:71]: this will result in string | undefined | null I believe, do you want string | null here instead?

File: source/frontend/src/features/documents/documentDetail/DocumentDetailForm.tsx [hunk 3]
```diff
@@ -58,6 +60,36 @@ export const DocumentDetailForm: React.FunctionComponent<
...
+    },
+    [props.documentTypes],
+  );
+
+  useEffect(() => {
```
Comment (devinleighsmith) [inline source/frontend/src/features/documents/documentDetail/DocumentDetailForm.tsx:76]: hmm, I'm not sure why this one is here, since this is a ref and won't trigger useEffect updates. Why is this necessary when the event handler on line 81 exists? Is this just a useEffect that is intended to fire once during init? If so, I'm not sure if a useEffect is necessary.

File: source/frontend/src/features/documents/documentDetail/DocumentDetailForm.tsx [hunk 4]
```diff
@@ -105,13 +137,18 @@ export const DocumentDetailForm: React.FunctionComponent<
...
-                        onChange={props.onDocumentTypeChange}
+                        onChange={onDocumentTypeChange}
                         disabled={
                           documentTypeOptions.length === 1 ||
                           props.relationshipType === ApiGen_CodeTypes_DocumentRelationType.Templates
                         }
                       />
                     </SectionField>
+                    {exists(documentTypePurpose) && (
+                      <SectionField label={null}>
```
Comment (devinleighsmith) [inline source/frontend/src/features/documents/documentDetail/DocumentDetailForm.tsx:148]: why label={null} instead of just no declaration?

**Review by devinleighsmith — APPROVED** [review]
approved with minor comments.

File: source/frontend/src/features/documents/documentDetail/DocumentDetailForm.tsx (hunk 3 above)
Comment (FuriousLlama) [reply source/frontend/src/features/documents/documentDetail/DocumentDetailForm.tsx:74]: Yes, its for the first render, otherwise the purpose is not set from the other function

File: source/frontend/src/features/documents/documentDetail/DocumentDetailForm.tsx (hunk 4 above)
Comment (FuriousLlama) [reply source/frontend/src/features/documents/documentDetail/DocumentDetailForm.tsx:146]: thats the way we have right now. The label is required on the section field. I like having no label instead of a required null
